day-range-picker/day-range-picker.js

```diff
@@ -0,0 +1,1050 @@
+"use strict";
+
+const hasEventListeners = !!window.addEventListener;
+const sto = window.setTimeout;
+const priceList = [
+  38, 27, 23, 24, 40, 28, 36, 41, 28, 22, 30, 45, 41, 35, 42, 28, 20, 22, 42,
+  29, 35, 39, 32, 21, 45, 43, 50, 24, 35, 25, 36, 33,
+];
+const defaults = {
+  // bind the picker to a form field
+  checkIn: null,
+  checkOut: null,
+  checkInLabel: null,
+  checkOutLabel: null,
+
+  priceList: null,
+
+  // automatically show/hide the picker on `field` focus (default `true` if `field` is set)
+  bound: undefined,
+
+  // position of the datepicker, relative to the field (default to bottom & left)
+  // ('bottom' & 'left' keywords are not used, 'top' & 'right' are modifier on the bottom/left position)
+  position: "bottom left",
+
+  // automatically fit in the viewport even if it means repositioning from the position option
+  reposition: true,
+
+  showPrice: false,
+
+  numberOfMonths: 2,
+
+  startDate: null,
+  
+  endDate: null,
+
+  // the default output format for `.toString()` and `field` value
+  format: "YYYY-MM-DD",
+
+  // the toString function which gets passed a current date object and format
+  // and returns a string
+  toString: null,
+
+  // used to create date object from current input string
+  parse: null,
+
+  // the minimum/earliest date that can be selected
+  minDate: null,
+  // the maximum/latest date that can be selected
+  maxDate: null,
+  minYear: 0,
+  maxYear: 9999,
+  minMonth: undefined,
+  maxMonth: undefined,
+
+  dpickerConst: {
+    previousMonth: "Previous Month",
+    nextMonth: "Next Month",
+    months: [
+      "January",
+      "February",
+      "March",
+      "April",
+      "May",
+      "June",
+      "July",
+      "August",
+      "September",
+      "October",
+      "November",
+      "December",
+    ],
+    monthsShort:[
+      "Jan",
+      "Feb",
+      "Mar",
+      "Apr",
+      "May",
+      "Jun",
+      "Jul",
+      "Aug",
+      "Sep",
+      "Oct",
+      "Nov",
+      "Dec",
+    ],
+    weekdays: [
+      "Sunday",
+      "Monday",
+      "Tuesday",
+      "Wednesday",
+      "Thursday",
+      "Friday",
+      "Saturday",
+    ],
+    weekdaysShort: ["Su", "Mo", "Tu", "We", "Th", "Fr", "Sa"],
+  },
+
+  // callback function
+  onSelect: null,
+  onOpen: null,
+  onClose: null,
+};
+const addEvent = function (el, e, callback, capture) {
+  if (hasEventListeners) {
+    el.addEventListener(e, callback, !!capture);
+  } else {
+    el.attachEvent("on" + e, callback);
+  }
+};
+
+const removeEvent = function (el, e, callback, capture) {
+  if (hasEventListeners) {
+    el.removeEventListener(e, callback, !!capture);
+  } else {
+    el.detachEvent("on" + e, callback);
+  }
+};
+
+const trim = function (str) {
+  return str.trim ? str.trim() : str.replace(/^\s+|\s+$/g, "");
+};
+
+const hasClass = function (el, cn) {
```

Comment:
Is there a reason you don't use the [classList](https://developer.mozilla.org/en-US/docs/Web/API/Element/classList) instead of `className` construct that has built-in methods (`add`, `remove`, `toggle`, `replace`) already?